### PR TITLE
fix: implement upgrade handler

### LINKED ISSUE #6285: Cosmos Upgrade Handler
## What is the Problem Being Solved?

To install Pismo on the chain, we need a stub upgrade handler that calls agoric-upgrade-8.

This helps drive the chain upgrade at a certain block height, even if the upgrade handler itself is empty.

### PR DESCRIPTION
**Pull request by arirubinstein**
closes: #6285
refs: #XXXX

## Description

Turns out since we're not starting from genesis, we need to tickle the InitGenesis on each of the newly added modules.  This change implements the upgrade handler.

### Testing Considerations

I tested this locally with a stock agoric-upgrade-7 local chain with no issues at the cosmos layer.   
Testing Scripts: https://github.com/Agoric/upgrade-test

```
11:52AM INF ABCI Replay Blocks appHeight=7 module=consensus stateHeight=7 storeHeight=8
11:52AM INF Replay last block using real app module=consensus
11:52AM INF applying upgrade "agoric-upgrade-8" at height: 8
11:52AM INF adding a new module: lien
2022/10/07 11:52:41 Running SwingSet until bootstrap is ready
Loading slog sender modules: @agoric/telemetry/src/flight-recorder.js
2022-10-07T18:52:41.327Z launch-chain: Launching SwingSet kernel
2022-10-07T18:52:54.474Z launch-chain: Expected SwingSet kernel statistic cosmic_swingset_inbound_queue_length not found
2022-10-07T18:52:54.474Z launch-chain: Expected SwingSet kernel statistic cosmic_swingset_inbound_queue_add not found
2022-10-07T18:52:54.474Z launch-chain: Expected SwingSet kernel statistic cosmic_swingset_inbound_queue_remove not found
2022-10-07T18:52:54.475Z block-manager: block bootstrap
11:53AM INF created new capability module=ibc name=ports/echo
11:53AM INF port binded module=x/ibc/port port=echo
11:53AM INF claimed capability capability=3 module=vibc name=ports/echo
11:53AM INF adding a new module: swingset
11:53AM INF adding a new module: vbank
11:53AM INF adding a new module: vibc
11:53AM INF adding a new module: vstorage
11:53AM INF pre-swingset upgrade params={"beans_per_unit":[{"beans":"800000000","key":"blockComputeLimit"},{"beans":"1000000000000","key":"feeUnit"},{"beans":"10000000000","key":"inboundTx"},{"beans":"1000000000","key":"message"},{"beans":"20000000","key":"messageByte"},{"beans":"200000000000","key":"minFeeDebit"},{"beans":"30000000","key":"vatCreation"},{"beans":"100","key":"xsnapComputron"}],"bootstrap_vat_config":"@agoric/vats/decentral-core-config.json","fee_unit_price":[{"amount":"1000000","denom":"uist"}],"power_flag_fees":[{"fee":[{"amount":"10000000","denom":"ubld"}],"power_flag":"SMART_WALLET"}],"queue_max":[{"key":"inbound","size":1000}]} subspace=swingset
11:53AM INF post-swingset upgrade params={"beans_per_unit":[{"beans":"6500000000","key":"blockComputeLimit"},{"beans":"1000000000000","key":"feeUnit"},{"beans":"10000000000","key":"inboundTx"},{"beans":"1000000000","key":"message"},{"beans":"20000000","key":"messageByte"},{"beans":"200000000000","key":"minFeeDebit"},{"beans":"30000000","key":"vatCreation"},{"beans":"100","key":"xsnapComputron"}],"bootstrap_vat_config":"@agoric/vats/decentral-prod-config.json","fee_unit_price":[{"amount":"1000000","denom":"uist"}],"power_flag_fees":[{"fee":[{"amount":"10000000","denom":"ubld"}],"power_flag":"SMART_WALLET"}],"queue_max":[{"key":"inbound","size":1000}]} subspace=swingset
11:53AM INF minted coins from module account amount=205stake from=mint module=x/bank
2022-10-07T18:53:05.091Z block-manager: block 8 begin
11:53AM INF executed block height=8 module=consensus num_invalid_txs=0 num_valid_txs=0
2022-10-07T18:53:05.116Z block-manager: block 8 commit
11:53AM INF commit synced commit=436F6D6D697449447B5B3135302037322035203234372032313520323239203137302031313420323335203138322036203134332038312036372031382036382032313420313937203131372032323020313831203220323030203230382037372031382031343020323620323520313534203631203138395D3A387D
11:53AM INF committed state app_hash=964805F7D7E5AA72EBB6068F51431244D6C575DCB502C8D04D128C1A199A3DBD height=8 module=consensus num_txs=0
11:53AM INF Completed ABCI Handshake - Tendermint and App are synced appHash="���\x00�<��\x17#��F�\\��c\u05eb��#�\\��>�yg�" appHeight=7 module=consensus
11:53AM INF Version info block=11 p2p=8 tendermint_version=0.34.19
11:53AM INF This node is a validator addr=2C1AB235A74EC6ACD7C287491A8857B95C82EAEB module=consensus pubKey=IwquZv0Eah0mnZgbEQx9yAbyXt2RQRqkUXpG+XTbz1I=
11:53AM INF indexed block height=8 module=txindex
11:53AM INF P2P Node ID ID=e2d0a862c4a6df88d153d66247c1506b40eea221 file=netstate/config/node_key.json module=p2p
11:53AM INF Adding persistent peers addrs=[] module=p2p
11:53AM INF Adding unconditional peer ids ids=[] module=p2p
11:53AM INF Add our address to book addr={"id":"e2d0a862c4a6df88d153d66247c1506b40eea221","ip":"0.0.0.0","port":26656} book=netstate/config/addrbook.json module=p2p
11:53AM INF Starting Node service impl=Node
11:53AM INF Starting pprof server laddr=localhost:6060
11:53AM INF Starting RPC HTTP server on 127.0.0.1:26657 module=rpc-server
11:53AM INF Starting P2P Switch service impl="P2P Switch" module=p2p
11:53AM INF Starting PEX service impl=PEX module=pex
11:53AM INF Starting AddrBook service book=netstate/config/addrbook.json impl=AddrBook module=p2p
11:53AM INF Starting Mempool service impl=Mempool module=mempool
11:53AM INF Starting BlockchainReactor service impl=BlockchainReactor module=blockchain
11:53AM INF Starting Consensus service impl=ConsensusReactor module=consensus
11:53AM INF Reactor  module=consensus waitSync=false
11:53AM INF Starting State service impl=ConsensusState module=consensus
11:53AM INF Ensure peers module=pex numDialing=0 numInPeers=0 numOutPeers=0 numToDial=10
11:53AM INF No addresses to dial. Falling back to seeds module=pex
11:53AM INF Starting baseWAL service impl=baseWAL module=consensus wal=netstate/data/cs.wal/wal
11:53AM INF Starting Group service impl=Group module=consensus wal=netstate/data/cs.wal/wal
11:53AM INF Starting TimeoutTicker service impl=TimeoutTicker module=consensus
11:53AM INF Searching for height height=9 max=0 min=0 module=consensus wal=netstate/data/cs.wal/wal
11:53AM INF Searching for height height=8 max=0 min=0 module=consensus wal=netstate/data/cs.wal/wal
11:53AM INF Found height=8 index=0 module=consensus wal=netstate/data/cs.wal/wal
11:53AM INF Catchup by replaying consensus messages height=9 module=consensus
11:53AM INF Replay: Done module=consensus
11:53AM INF Starting Evidence service impl=Evidence module=evidence
11:53AM INF Starting StateSync service impl=StateSync module=statesync
11:53AM INF Saving AddrBook to file book=netstate/config/addrbook.json module=p2p size=0
11:53AM INF Timed out dur=4998.431 height=9 module=consensus round=0 step=1
11:53AM INF received proposal module=consensus proposal={"Type":32,"block_id":{"hash":"492A4F3054B20A8AA0E3C798BE07463604A2FB73D86531D1AC125F69425F6B93","parts":{"hash":"6FC5DAD0F32DFDE4BD9B948C57CCEF89903BCA27FCF0314C4F0A60D201A352A9","total":1}},"height":9,"pol_round":-1,"round":0,"signature":"/EqpdZqoOp054CTmZSH1vCMkL31PA42NcuO08dFzlz5jsXWWHYeVfZB+J2Fkx/WWurmfps2M/4RfHX7Lq8QaAw==","timestamp":"2022-10-07T18:53:10.297537Z"}
11:53AM INF received complete proposal block hash=492A4F3054B20A8AA0E3C798BE07463604A2FB73D86531D1AC125F69425F6B93 height=9 module=consensus
11:53AM INF finalizing commit of block hash=492A4F3054B20A8AA0E3C798BE07463604A2FB73D86531D1AC125F69425F6B93 height=9 module=consensus num_txs=0 root=964805F7D7E5AA72EBB6068F51431244D6C575DCB502C8D04D128C1A199A3DBD
11:53AM INF minted coins from module account amount=205stake from=mint module=x/bank
2022-10-07T18:53:10.472Z block-manager: block 9 begin
11:53AM INF executed block height=9 module=state num_invalid_txs=0 num_valid_txs=0
2022-10-07T18:53:10.496Z block-manager: block 9 commit
11:53AM INF commit synced commit=436F6D6D697449447B5B313931203233322038342031343420313839203634203439203234382031353220313536203833203234332033382039332031332032313320323136203232342032353220383320323039203130382031393720323133203234372031342032343620323032203233342032303320323335203233385D3A397D
11:53AM INF committed state app_hash=BFE85490BD4031F8989C53F3265D0DD5D8E0FC53D16CC5D5F70EF6CAEACBEBEE height=9 module=state num_txs=0

```

### CODE REVIEW

File: golang/cosmos/app/app.go [hunk 1]
```diff
@@ -732,10 +733,47 @@ func NewAgoricApp(
...
 		func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
-			return vm, nil
+			//Run migrations first so InitGenesis is called for swingset, vibc, vbank, vstorage, lien
+			vm, err := app.mm.RunMigrations(ctx, app.configurator, vm)
+			if err != nil {
+				return vm, err
+			}
+
+			// Set bean count
+			currentSwingset := app.SwingSetKeeper.GetParams(ctx)
```
Comment (JimLarson) [inline golang/cosmos/app/app.go]: Consider renaming as `swingParams` or something with "params" in it.

File: golang/cosmos/app/app.go [hunk 2]
```diff
@@ -732,10 +733,47 @@ func NewAgoricApp(
...
+			ctx.Logger().Info("pre-swingset upgrade", "subspace", swingset.ModuleName, "params", currentSwingset)
+
+			for i := 0; i < len(currentSwingset.BeansPerUnit); i++ {
+				if currentSwingset.BeansPerUnit[i].Key == swingsettypes.BeansPerBlockComputeLimit {
+					currentSwingset.BeansPerUnit[i] = swingsettypes.NewStringBeans(swingsettypes.BeansPerBlockComputeLimit, sdk.NewUint(6500000000))
```
Comment (JimLarson) [inline golang/cosmos/app/app.go]: I think you can just do

```
currentSwingset.BeansPerUnit[i].Beans = sdk.NewUint(6_500_000_000)
```

**Review by JimLarson — APPROVED** [review]
Golang code LGTM with some optional improvements.

I can't speak to the correctness of the JSON contents - please get separate approval for that.

**Comment by arirubinstein** [comment]
Adding @dckc for the stub prod bootstrap

**Comment by dckc** [comment]
Did your test include starting swingset and the PSM and all that?

**Comment by arirubinstein** [comment]
I didn't test swingset interactively, however there was state populated in vstorage, etc in an export:
```
    "vstorage": {
      "data": [
        {
          "path": "activityhash",
          "value": "71ad763bbcc59d03f6b5dd3a7700674a383c2594db03f8a9bebc19d2bb3fb5f6"
        },
        {
          "path": "published.agoricNames.brand",
          "value": "{\"blockHeight\":\"19\",\"values\":[\"{\\\"body\\\":\\\"[]\\\",\\\"slots\\\":[]}\",\"{\\\"body\\\":\\\"[[\\\\\\\"BLD\\\\\\\",{\\\\\\\"@qclass\\\\\\\":\\\\\\\"slot\\\\\\\",\\\\\\\"iface\\\\\\\":\\\\\\\"Alleged: BLD brand\\\\\\\",\\\\\\\"index\\\\\\\":0}]]\\\",\\\"slots\\\":[\\\"board0592\\\"]}\",\"{\\\"body\\\":\\\"[[\\\\\\\"BLD\\\\\\\",{\\\\\\\"@qclass\\\\\\\":\\\\\\\"slot\\\\\\\",\\\\\\\"iface\\\\\\\":\\\\\\\"Alleged: BLD brand\\\\\\\",\\\\\\\"index\\\\\\\":0}],[\\\\\\\"IST\\\\\\\",{\\\\\\\"@qclass\\\\\\\":\\\\\\\"slot\\\\\\\",\\\\\\\"iface\\\\\\\":\\\\\\\"Alleged: IST brand\\\\\\\",\\\\\\\"index\\\\\\\":1}]]\\\",\\\"slots\\\":[\\\"board0592\\\",\\\"board0074\\\"]}\"]}"
        },
        {
          "path": "published.agoricNames.installation",
          "value": "{\"blockHeight\":\"19\",\"values\":[\"{\\\"body\\\":\\\"[]\\\",\\\"slots\\\":[]}\"]}"
        },
        {
          "path": "published.agoricNames.instance",
          "value": "{\"blockHeight\":\"19\",\"values\":[\"{\\\"body\\\":\\\"[]\\\",\\\"slots\\\":[]}\"]}"
        },
        {
          "path": "published.agoricNames.issuer",
          "value": "{\"blockHeight\":\"19\",\"values\":[\"{\\\"body\\\":\\\"[]\\\",\\\"slots\\\":[]}\",\"{\\\"body\\\":\\\"[[\\\\\\\"IST\\\\\\\",{\\\\\\\"@qclass\\\\\\\":\\\\\\\"slot\\\\\\\",\\\\\\\"iface\\\\\\\":\\\\\\\"Alleged: IST issuer\\\\\\\",\\\\\\\"index\\\\\\\":0}]]\\\",\\\"slots\\\":[\\\"board0371\\\"]}\",\"{\\\"body\\\":\\\"[[\\\\\\\"BLD\\\\\\\",{\\\\\\\"@qclass\\\\\\\":\\\\\\\"slot\\\\\\\",\\\\\\\"iface\\\\\\\":\\\\\\\"Alleged: BLD issuer\\\\\\\",\\\\\\\"index\\\\\\\":0}],[\\\\\\\"IST\\\\\\\",{\\\\\\\"@qclass\\\\\\\":\\\\\\\"slot\\\\\\\",\\\\\\\"iface\\\\\\\":\\\\\\\"Alleged: IST issuer\\\\\\\",\\\\\\\"index\\\\\\\":1}]]\\\",\\\"slots\\\":[\\\"board0223\\\",\\\"board0371\\\"]}\"]}"
        },
        {
          "path": "published.agoricNames.oracleBrand",
          "value": "{\"blockHeight\":\"19\",\"values\":[\"{\\\"body\\\":\\\"[]\\\",\\\"slots\\\":[]}\"]}"
        },
        {
          "path": "published.agoricNames.uiConfig",
          "value": "{\"blockHeight\":\"19\",\"values\":[\"{\\\"body\\\":\\\"[]\\\",\\\"slots\\\":[]}\"]}"
        }
      ]
    }

```

Going to test this live in a network soon

**Review by dckc — APPROVED** [review]
looks good.

`diff -ur decentral-psm-config.json decentral-prod-config.json` shows no diffs.

**Pull request merged**